mplc/scenario.py

```diff
@@ -304,13 +304,21 @@ def __init__(
 
         now = datetime.datetime.now()
         now_str = now.strftime("%Y-%m-%d_%Hh%M")
-        self.scenario_name = f"scenario_{self.scenario_id}_repeat_{self.n_repeat}_{now_str}_" \
-                             f"{uuid.uuid4().hex[:3]}"  # to distinguish identical names
-        self.short_scenario_name = f"{self.partners_count} {self.amounts_per_partner}"
-
-        self.experiment_path = experiment_path
-        self.save_folder = self.experiment_path / self.scenario_name
-
+        self.scenario_name = kwargs.get('scenario_name',
+                                        f"scenario_{self.scenario_id}_repeat_{self.repeat_count}_{now_str}_"
+                                        f"{uuid.uuid4().hex[:3]}")  # to distinguish identical names
+        if ' ' in self.scenario_name:
+            raise ValueError(
+                f'The scenario name "{self.scenario_name}"cannot be written with space character, please use '
+                f'underscore or dash.')
+        self.short_scenario_name = f"{self.partners_count}_{self.amounts_per_partner}"
+
+        if save_path is not None:
+            self.experiment_path = Path(save_path)
+            self.save_folder = self.experiment_path / self.scenario_name
+        else:
+            self.experiment_path = None
+            self.save_folder = None
```

Comment:
Hmmm, so a `scenario` instance gets an `experiment_path` attribute? A bit weird? To me either it is added to an experiment, in which case the experiment `experiment_path` attribute exists, or it is not and in that case it is useless. What did I miss?